packages/aws-cdk-lib/aws-cloudfront/lib/realtime-log-config.ts

```diff
@@ -0,0 +1,144 @@
+    return new (class extends Endpoint {
+      public _renderEndpoint(scope: Construct) {
+        const cloudfrontRole = role ?? singletonKinesisRole(scope);
+
+        stream.grant(cloudfrontRole,
+          'kinesis:DescribeStreamSummary',
+          'kinesis:DescribeStream',
+          'kinesis:PutRecord',
+          'kinesis:PutRecords',
+        );
+
+        if (stream.encryptionKey) {
+          stream.encryptionKey.grant(cloudfrontRole, 'kms:GenerateDataKey');
+        }
+
+        return {
+          kinesisStreamConfig: {
+            roleArn: cloudfrontRole.roleArn,
+            streamArn: stream.streamArn,
+          },
+          streamType: 'Kinesis',
+        };
+      }
+    });
+  }
+
+  private constructor() {}
+
+  /**
+  * @internal
+  */
+  public abstract _renderEndpoint(scope: Construct): any;
+}
+
+/**
+ * Represents Realtime Log Configuration
+ */
+export interface IRealtimeLogConfig extends IResource {
+  /**
+   * The name of the realtime log config.
+   * @attribute
+   */
+  readonly realtimeLogConfigName: string;
+  /**
+   * The arn of the realtime log config.
+   * @attribute
+   */
+  readonly realtimeLogConfigArn: string;
+}
+
+/**
+ * Properties for defining a RealtimeLogConfig resource.
+ */
+export interface RealtimeLogConfigProps {
+  /**
+   * The unique name of this real-time log configuration.
+   *
+   * @default - the unique construct ID
+   */
+  readonly realtimeLogConfigName?: string;
+  /**
+   * A list of fields that are included in each real-time log record.
+   *
+   * @see https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/real-time-logs.html#understand-real-time-log-config-fields
+   */
+  readonly fields: string[];
+  /**
+   * The sampling rate for this real-time log configuration.
+   */
+  readonly samplingRate: number;
+  /**
+   * Contains information about the Amazon Kinesis data stream where you are sending real-time log data for this real-time log configuration.
+   */
+  readonly endPoints: Endpoint[];
+}
+
+/**
+ * A Realtime Log Config configuration
+ *
+ * @resource AWS::CloudFront::RealtimeLogConfig
+ */
+export class RealtimeLogConfig extends Resource implements IRealtimeLogConfig {
+  public readonly realtimeLogConfigName: string;
+  public readonly realtimeLogConfigArn: string;
+
+  constructor(scope: Construct, id: string, props: RealtimeLogConfigProps) {
+    super(scope, id, {
+      physicalName: props.realtimeLogConfigName ?? Lazy.string({ produce: () => Names.uniqueId(this) }),
+    });
+
+    if ((props.samplingRate < 1 || props.samplingRate > 100)) {
+      throw new Error(`Sampling rate must be between 1 and 100 (inclusive), received ${props.samplingRate}`);
+    }
+
+    const resource = new CfnRealtimeLogConfig(this, 'Resource', {
+      endPoints: props.endPoints.map(endpoint => {
+        return endpoint._renderEndpoint(this);
+      }),
+      fields: props.fields,
```

Comment:
Have we considered users that may want to add `fields` or `endpoints` after creating the `RealtimeLogConfig`? This could be done with something like `addEndpoint` or `addField`. Note that these two properties would then need to be evaluated with `Lazy`

Reply:
I can't see it being used afterward `RealtimeLogConfig` - is it something we could come back to as a feature if required/asked for?

Reply:
Yeah that's fair.